src/Microsoft.NetCore.Analyzers/Core/Performance/DoNotUseCountWhenAnyCanBeUsed.Fixer.cs

```diff
@@ -52,57 +58,72 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
             var node = root.FindNode(context.Span);
+            var isAsync = context.Diagnostics[0].Id == DoNotUseCountWhenAnyCanBeUsedAnalyzer.AsyncRuleId;
 
             if (node is object &&
-                this.TryGetFixer(node, out var expression, out var arguments, out var negate))
+                this.TryGetFixer(node, isAsync, out var expression, out var arguments, out var negate))
             {
                 context.RegisterCodeFix(
-                    new DoNotUseCountWhenAnyCanBeUsedCodeAction(context.Document, node, expression, arguments, negate),
+                    new DoNotUseCountAsyncWhenAnyAsyncCanBeUsedCodeAction(isAsync, context.Document, node, expression, arguments, negate),
                     context.Diagnostics);
             }
         }
 
         /// <summary>
-        /// Tries the get a fixer the specified <paramref name="node"/>.
+        /// Tries the get a fixer the specified <paramref name="node" />.
         /// </summary>
         /// <param name="node">The node to get a fixer for.</param>
-        /// <param name="expression">If this method returns <see langword="true"/>, contains the expression to be used to invoke <c>Any</c>.</param>
-        /// <param name="arguments">If this method returns <see langword="true"/>, contains the arguments from <c>Any</c> to be used on <c>Count</c>.</param>
-        /// <param name="negate">If this method returns <see langword="true"/>, indicates whether to negate the expression.</param>
+        /// <param name="isAsync"><see langword="true" /> if it's an asynchronous method; <see langword="false"/> otherwise.</param>
+        /// <param name="expression">If this method returns <see langword="true" />, contains the expression to be used to invoke <c>Any</c>.</param>
+        /// <param name="arguments">If this method returns <see langword="true" />, contains the arguments from <c>Any</c> to be used on <c>Count</c>.</param>
+        /// <param name="negate">If this method returns <see langword="true" />, indicates whether to negate the expression.</param>
         /// <returns><see langword="true" /> if a fixer was found., <see langword="false" /> otherwise.</returns>
-        protected abstract bool TryGetFixer(SyntaxNode node, out SyntaxNode expression, out IEnumerable<SyntaxNode> arguments, out bool negate);
+        protected abstract bool TryGetFixer(SyntaxNode node, bool isAsync, out SyntaxNode expression, out IEnumerable<SyntaxNode> arguments, out bool negate);
 
-        private class DoNotUseCountWhenAnyCanBeUsedCodeAction : CodeAction
+        private class DoNotUseCountAsyncWhenAnyAsyncCanBeUsedCodeAction : CodeAction
```

Comment:
Given this code action is used for fixing both CA1827 and CA1828, we should probably just use a terminology that may imply it is used only for one of them. Even something like `MyCodeAction` used across the repo would be fine, or any other better neutral term.